### PR TITLE
hotfeature/8440 - add new route for change password page

### PR DESCRIPTION
**Pull request by ft-tiny**
add new route for change password page
https://jira.fluxtech.me/browse/THICKETFAI-8440

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -268,3 +268,5 @@ export const postReferralVisits = (eventId: string, referralId: string) =>
 export const createAccount = (data: any) => publicRequest.post(`/v1/oauth/register-rn`, data)
 // forgot password
 export const forgotPassword = (email: string) => publicRequest.post(`/v1/oauth/restore-password-rn`, { email })
+// reset password
+export const resetPassword = (data: any) => publicRequest.post(`/v1/oauth/reset-password-rn`, data)
```
Comment (ft-worker) [inline src/api/index.ts:272]: Please add data type.

File: example/Routes.tsx [hunk 2]
```diff
@@ -133,5 +134,12 @@ export const ROUTES = (
       />
     } />
     <Route path="/order" element={<OrderDetailsContainer />} />
+    <Route path="/reset-password" element={
+      <ResetPasswordContainer
+        tokenProps=''
```
Comment (ft-worker) [inline example/Routes.tsx:139]: Rename `tokenProps`. It's `token` property getting from props.

File: src/components/resetPasswordContainer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                      helperText={meta.touched && meta.error}
+                      {...field}
+                    />
+                  )}
+                </Field>
+              </div>
+              <div className="field-box">
+                <Field name="confirmPassword">
+                  {({ field, meta }: { [key: string]: any }) => (
+                    <TextField
```
Comment (ft-worker) [inline src/components/resetPasswordContainer/index.tsx:83]: Use `CustomField` component.

File: src/components/resetPasswordContainer/index.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          }
+        }}
+      >
+        {({ isValid, dirty }) => (
+          <Form>
+            <div className="body">
+              <div className="field-box">
+                <Field name="password">
+                  {({ field, meta }: { [key: string]: any }) => (
+                    <TextField
```
Comment (ft-worker) [inline src/components/resetPasswordContainer/index.tsx:69]: Use `CustomField` component.

File: src/components/resetPasswordContainer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
+import React, { FC, useState } from 'react'
+import { Field, Form, Formik } from 'formik'
+import { TextField, CircularProgress } from '@mui/material'
+import axios, { AxiosError } from 'axios'
+import * as Yup from 'yup'
+import { resetPassword } from '../../api'
+import './style.css'
+
+interface Props {
```
Comment (ft-worker) [inline src/components/resetPasswordContainer/index.tsx:9]: Please prefix interface with `I` and rename it: `IResetPasswordProps` or `IResetPasswordContainerProps`.

File: src/components/resetPasswordContainer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            }
+
+            const payload = {
+              token,
+              ...values,
+            }
+            const res = await resetPassword(payload)
+            onResetPasswordSuccess(res)
+          } catch (e) {
+            if (axios.isAxiosError(e)) onResetPasswordError(e)
```
Comment (ft-worker) [inline src/components/resetPasswordContainer/index.tsx:57]: Add `if` condition's body brackets.

**Review by ft-worker — CHANGES_REQUESTED** [review]

**Comment by ft-worker** [comment]
@ft-tiny add task's description and number in PR's description.

File: src/api/index.ts (hunk 1 above)
Comment (ft-tiny) [reply src/api/index.ts]: done

File: example/Routes.tsx (hunk 2 above)
Comment (ft-tiny) [reply example/Routes.tsx]: done

File: src/components/resetPasswordContainer/index.tsx (hunk 3 above)
Comment (ft-tiny) [reply src/components/resetPasswordContainer/index.tsx]: done

File: src/components/resetPasswordContainer/index.tsx (hunk 4 above)
Comment (ft-tiny) [reply src/components/resetPasswordContainer/index.tsx]: done

File: src/components/resetPasswordContainer/index.tsx (hunk 5 above)
Comment (ft-tiny) [reply src/components/resetPasswordContainer/index.tsx]: done

File: src/components/resetPasswordContainer/index.tsx (hunk 6 above)
Comment (ft-tiny) [reply src/components/resetPasswordContainer/index.tsx]: done

File: src/components/resetPasswordContainer/index.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+          } finally {
+            setLoading(false)
+          }
+        }}
+      >
+        {({ isValid, dirty }) => (
+          <Form>
+            <div className="body">
+              <div className="field-item">
+                <Field
```
Comment (ft-worker) [inline src/components/resetPasswordContainer/index.tsx:70]: Field type must be `password`.

File: src/components/resetPasswordContainer/index.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+                  name="password"
+                  label="New Password"
+                  component={CustomField}
+                />
+              </div>
+              <div className="field-item">
+                <Field
```
Comment (ft-worker) [inline src/components/resetPasswordContainer/index.tsx:77]: Field type must be `password`.

File: src/components/resetPasswordContainer/index.tsx (hunk 7 above)
Comment (ft-tiny) [reply src/components/resetPasswordContainer/index.tsx:70]: done

**Review by ft-worker — APPROVED** [review]

**Pull request merged**